### PR TITLE
Refactor loans

### PR DESCRIPTION
**Pull request by 0xjocke**
`import { tx } from 'utils/transactions';` had some bugs in it. Rather than fixing that I refactored all the loans action to use `useSynthetixTxn`.

### CODE REVIEW

File: sections/loans/components/ActionBox/ActiveBorrowsTab/ModifyLoanActions/Deposit.tsx [hunk 1]
```diff
@@ -54,94 +52,77 @@ const Deposit: React.FC<DepositProps> = ({
...
-		[isApproved, getApproveTxData, getDepositTxData]
+	const isApprovedQuery = useQuery(
+		'isApproved',
+		async () => {
+			const allowance = await collateralAssetContract.allowance(walletAddress, loanContractAddress);
+			return wei(allowance).gte(depositAmount);
+		},
+		{ enabled: Boolean(collateralAssetContract && loanContractAddress && walletAddress) }
+	);
+	const isApproved = loanTypeIsETH ? true : isApprovedQuery.data;
```
Comment (jmzwar) [inline sections/loans/components/ActionBox/ActiveBorrowsTab/ModifyLoanActions/Deposit.tsx:63]: loadTypeIsETH || isApprovedQuery.data could also work here.

**Review by jmzwar — APPROVED** [review]
Looks good but I will need to do some testing.

**Review by noisekit — APPROVED** [review]
LGTM!

File: sections/loans/components/ActionBox/ActiveBorrowsTab/ModifyLoanActions/Deposit.tsx [hunk 2]
```diff
@@ -54,94 +52,77 @@ const Deposit: React.FC<DepositProps> = ({
...
 	);
 
-	const onApproveOrDeposit = async () => {
-		setError(null);
-		!isApproved ? approve() : deposit();
-	};
+	const contractName = loanTypeIsETH ? 'CollateralEth' : 'CollateralErc20';
+
+	const depositTxn = useSynthetixTxn(
+		contractName,
```
[the remark this answers is not in the document]
Comment (bachstatter) [reply sections/loans/components/ActionBox/ActiveBorrowsTab/ModifyLoanActions/Deposit.tsx:85]: yea some good points..

File: sections/loans/components/ActionBox/ActiveBorrowsTab/ModifyLoanActions/Deposit.tsx (hunk 1 above)
Comment (bachstatter) [reply sections/loans/components/ActionBox/ActiveBorrowsTab/ModifyLoanActions/Deposit.tsx:63]: This logic will be removed in #1195 so not doing anything related to this

**Pull request merged**